Add support for inline '//' and '/* */' comments

mvparker810 opened the pull request:
As noted in armv8-examples/documentation/armsim_guide.md, the parser did not support inline comments. This commit adds that support.

Examples that now work:
	mov x0, #0 // hello world
	mov x1, #0 /* hello world */

mvparker810 committed:
Add support for inline '//' and '/* */' comments

As noted in armv8-examples/documentation/armsim_guide.md, the parser did not support inline comments. This commit adds that support.

Examples that now work:
	mov x0, #1 // hello world
	mov x1, #0 /* hello world */

mvparker810 committed:
Add support for inline '//' and '/* */' comments

As noted in armv8-examples/documentation/armsim_guide.md, the parser did not support inline comments. This commit adds that support.

Examples that now work:
	mov x0, #1 // hello world
	mov x1, #0 /* hello world */

d-gaston commented:
Thanks for the pull request! It looks good to me. Just one thing though, can you add a new file in the tests/ directory for this? Basically the example you put in your comment above will suffice but you can add more test cases if you think of them.

mvparker810 committed:
more robust comment handling/testing

Added 'tests/comments_test.s', which is functionally identical to 'arithmetic_test.s' but includes extensive comments to rigorously test comment handling.

While writing tests, I discovered even more edge cases that I hadn't considered. I refactored all comment stripping logic into a dedicated 'remove_comments()' preprocessor function that runs before parsing: to simplify the main parsing logic and ensure consistent handling of comment patterns.

I also removed the comment documentation in 'armsim.py' and 'armsim_guide.md' to be accurate with these changes.

mvparker810 committed:
resolved merge conflict

d-gaston commented:
Looks good, thanks!
